utils/reports.ts

```diff
@@ -0,0 +1,92 @@
+import * as FileSystem from 'expo-file-system';
+import * as Sharing from 'expo-sharing';
+import moment from 'moment';
+import { CatSterilizationReport, castToReport } from '../models/Report';
+import { isWeb } from '../utils/platform';
+import SnackbarManager from './SnackbarManager';
+import { makeRequest } from './server';
+
+export const getReports = async (
+  checkInDate: Date,
+  checkOutDate: Date
+): Promise<{
+  success: boolean;
+  report?: CatSterilizationReport;
+}> => {
+  const formattedCheckInDate = moment(checkInDate).format('YYYY-MM-DD');
+  const formattedCheckOutDate = moment(checkOutDate).format('YYYY-MM-DD');
+
+  const { error, data } = await makeRequest({
+    path: `/report?check_in_date=${formattedCheckInDate}&check_out_date=${formattedCheckOutDate}`,
+    method: 'GET',
```

Comment:
They are technically initialised with the current date, so I think this is not an issue. However, as you pointed out, we should probably prevent the user from selecting a start date that is greater than the end date, since this is an inconsistency that we didn't consider yet. Fortunately, selecting a start date greater than the end date doesn't result in any errors, but it might be useful to inform the user about this with a simple snackbar message.

Reply:
By the way, I also checked if editing the date manually (by editing the text) and setting a non-sensical value breaks the app in any way e.g. tapping on the pencil icon and changing `07/02/2024` to `0x12345`. It really doesn't break the app, but we could add a message for this as well (if it is even possible to display a snackbar within this component). I will note this down so that we can include it later on.